sensor/sensor_calibration_manager/sensor_calibration_manager/sensor_calibration_manager.py

```diff
@@ -0,0 +1,317 @@
+
+    def on_calibration_finished(self):
+        tf_dict = self.calibrator.get_calibration_results()
+        processed_tf_dict = self.calibrator.get_processed_calibration_results()
+        self.calibration_tfs_view.setTfs(tf_dict)
+
+        # There are two main cases:
+        # 1) The processed_tf_dict modify the original tfs
+        # 2) The processed tf_dict makes a new structure, hence it must be displayed as such
+        final_tfs_dict = copy.deepcopy(self.tfs_dict)
+
+        changed_frames_dict = self.calibration_results = defaultdict(set)
+
+        for parent, child_and_tf in processed_tf_dict.items():
+            if parent not in final_tfs_dict:
+                continue
+            for (
+                child,
+                transform,
+            ) in child_and_tf.items():
+                if child in final_tfs_dict[parent]:
+                    final_tfs_dict[parent][child] = transform
+                    changed_frames_dict[parent].add(child)
+
+        if len(changed_frames_dict) == len(processed_tf_dict):
+            self.final_tfs_view.setTfs(
+                final_tfs_dict,
+                changed_frames_dict=changed_frames_dict,
+                required_frames=self.calibrator.required_frames,
+            )
+        else:
+            self.final_tfs_view.setTfs(
+                processed_tf_dict,
+                changed_frames_dict=None,
+                required_frames=self.calibrator.required_frames,
+            )
+
+    def on_save_request(self):
+        logging.debug("on_save_request")
+
+        output_path = QFileDialog.getSaveFileName(
+            None, "Save File", f"{os.getcwd()}/calibration_results.yaml", "YAML files (*.yaml)"
+        )
+
+        logging.debug(output_path)
+        output_path = output_path[0]
+
+        if output_path is None or output_path == "":
+            logging.error("Invalid path")
+            return
+
+        logging.info(f"Saving calibration results to {output_path}")
+        self.calibrator.save_results(output_path, use_rpy=True)
+
+    def tfs_graph_callback(self, tfs_dict):
+        with self.lock:
+            tfs_dict = copy.deepcopy(tfs_dict)
+
+        self.tfs_graph_signal.emit(copy.deepcopy(tfs_dict))
+
+    def tf_graph_callback2(self, tfs_dict):
+        if self.calibrator.state != CalibratorState.WAITING_TFS:
+            return
+
+        for parent, children_and_tf_dict in tfs_dict.items():
+            self.tfs_dict[parent].update(children_and_tf_dict)
+
+        if self.tfs_dict and len(self.tfs_dict) > 0 and self.calibrator:
+            self.initial_tfs_view.setTfs(
+                self.tfs_dict, required_frames=self.calibrator.required_frames
+            )
+
+
+def main(args=None):
+    os.environ["QT_QPA_PLATFORM_PLUGIN_PATH"] = ""
+    app = QApplication(sys.argv)
+
+    rclpy.init(args=args)
+
+    try:
+        signal.signal(signal.SIGINT, sigint_handler)
+        calibration_manager = NewExtrinsicCalibrationManager()  # noqa: F841
```

Comment:
Change to ExtrinsicCalibrationManager

Reply:
Addressed in  d33e95d
The correct name is SensorCalibrationManager though